fs/inode/fs_files.c

```diff
@@ -51,59 +52,118 @@
  * Private Functions
  ****************************************************************************/
 
+/****************************************************************************
+ * Name: files_fget_by_index
+ ****************************************************************************/
+
+static FAR struct file *files_fget_by_index(FAR struct filelist *list,
+                                         int l1, int l2)
+{
+  FAR struct file *filep;
+  irqstate_t flags;
+
+  flags = spin_lock_irqsave(NULL);
```

Comment:
how about change fl_lock from mutex to spinlock to improve the efficient in SMP?

Reply:
Can add a lock only for file?

Reply:
Done